src/neo/Consensus/ConsensusService.cs

```diff
@@ -174,11 +177,20 @@ private void InitializeConsensus(byte viewNumber)
                 }
                 else
                 {
+                    // Too long (if changeview happens during consensus) or too short (which will occur during synchronization) 
+                    // consensus period will be filtered. Only normal consensus should be taken into account.
+                    if (block_received_gap > Blockchain.TimePerBlock / 3 && block_received_gap < Blockchain.TimePerBlock * 2)
+                    {
+                        // Renew time per block by last consensus period.
+                        block_span = (8 * block_span + 3 * (2 * Blockchain.TimePerBlock - block_received_gap)) / 11.0;
+                    }
```

Comment:
else block_span = Blockchain.TimePerBlock; ?

I think that we should be able to remove the variable https://github.com/neo-project/neo/pull/1593/files#diff-98ea4c9a410746331ba188dc2f45afadR32 and compute here according to block_received_gap

Reply:
If we use `block_span = Blockchain.TimePerBlock;` we will not be able to adjust time span. We use variable `block_span` to compensate dynamically if last `block_received_gap` exceeds or doesn't reach `Blockchain.TimePerBlock`